File: javaee/persistence/appengine-spring/src/main/java/org/sgodden/tom/domain/CollectionDetails.java
package org.sgodden.tom.domain;

import com.google.appengine.api.datastore.Key;
import java.io.Serializable;
import javax.jdo.annotations.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

/**
 */
@Entity
public class CollectionDetails implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Key id;

    @Embedded
    private Address address;

    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }
}

File: templates/scala-mongodb/model/src/main/scala/org/sgodden/tom/model/CollectionDetails.scala
package org.sgodden.tom.model

import com.novus.salat.annotations.raw.Persist

case class CollectionDetails() extends ICollectionDetails {

  @Persist
  private var address: Address = null

  def getAddress = address
  def setAddress(address: IAddress) = this.address = address.asInstanceOf[Address]

}
